node/cli/src/service.rs

```diff
@@ -77,38 +70,29 @@ construct_service_factory! {
 			{ |config: FactoryFullConfiguration<Self>, executor: TaskExecutor|
 				FullComponents::<Factory>::new(config, executor) },
 		AuthoritySetup = {
-			|mut service: Self::FullService, executor: TaskExecutor, key: Option<Arc<Pair>>| {
+			|mut service: Self::FullService, executor: TaskExecutor, local_key: Option<Arc<Pair>>| {
 				let (block_import, link_half) = service.config.custom.grandpa_import_setup.take()
 					.expect("Link Half and Block Import are present for Full Services or setup failed before. qed");
 
-				let local_key = if let Some(key) = key {
-					if !service.config.custom.grandpa_authority_only {
-						info!("Using authority key {}", key.public());
-						let proposer = Arc::new(substrate_service::ProposerFactory {
-							client: service.client(),
-							transaction_pool: service.transaction_pool(),
-						});
-
-						let client = service.client();
-						executor.spawn(start_aura(
-							SlotDuration::get_or_compute(&*client)?,
-							key.clone(),
-							client,
-							block_import.clone(),
-							proposer,
-							service.network(),
-						));
-					}
-
-					if service.config.custom.grandpa_authority {
-						info!("Running Grandpa session as Authority {}", key.public());
-						Some(key)
-					} else {
-						None
-					}
-				} else {
-					None
-				};
+				if let Some(ref key) = local_key {
+					info!("Using authority key {}", key.public());
+					let proposer = Arc::new(substrate_service::ProposerFactory {
+						client: service.client(),
+						transaction_pool: service.transaction_pool(),
+					});
+
+					let client = service.client();
+					executor.spawn(start_aura(
+						SlotDuration::get_or_compute(&*client)?,
+						key.clone(),
+						client,
+						block_import.clone(),
+						proposer,
+						service.network(),
+					));
+
+					info!("Running Grandpa session as Authority {}", key.public());
+				}
 
 				let voter = grandpa::run_grandpa(
 					grandpa::Config {
```

Comment:
When given `None` as `local_key` the grandpa voter runs the protocol but doesn't broadcast any messages.